### PR TITLE
Add CRUD tests for operating system entity for API part

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/api/test_operatingsystem.py
.S.................................
----------------------------------------------------------------------
Ran 35 tests in 208.382s

OK (SKIP=1)
```

### CODE REVIEW

**Comment by oshtaier** [comment]
Closes #3056

File: tests/foreman/api/test_operatingsystem.py [hunk 1]
```diff
@@ -50,15 +53,101 @@ def test_bz_1114640(self):
...
+    @run_only_on('sat')
+    @skip_if_bug_open('bugzilla', 1230902)
+    def test_create_with_minor_1230902(self):
+        """@Test: Create an operating system with an integer minor version.
+
+        @Feature: Operating System
+
+        @Assert: The minor version can be read back as a string.
+        """
+        minor = gen_integer(0, 9999999999999999)  # max len: 16 chars
```
Comment (abalakh) [inline tests/foreman/api/test_operatingsystem.py]:
```python
minor = gen_string('numeric', gen_integer(1, 16))
```

Comment (oshtaier) [inline tests/foreman/api/test_operatingsystem.py]: Sorry, but
`gen_string('numeric')` is not equal to `gen_integer`. That test was created to test integer value explicitly

Comment (abalakh) [inline tests/foreman/api/test_operatingsystem.py]: Cool, why not
```python
entities.OperatingSystem(minor=int(minor)).create()
```
then?
Hardcoding 16-digits value doesn't look that good, imo

File: tests/foreman/api/test_operatingsystem.py [hunk 2]
```diff
@@ -50,15 +53,101 @@ def test_bz_1114640(self):
...
+    @run_only_on('sat')
+    def test_positive_create_with_description(self):
+        """@Test: Create operating system with description
+
+        @Feature: Operating System
+
+        @Assert: Operating system entity is created and has proper description
+        """
+        name = gen_string('utf8')
+        desc = gen_string('utf8')
```
Comment (abalakh) [inline tests/foreman/api/test_operatingsystem.py]: Could you please check different values with subTest here?

File: tests/foreman/api/test_operatingsystem.py [hunk 3]
```diff
@@ -66,14 +155,32 @@ def test_point_to_arch(self):
...
+    def test_positive_create_with_multiple_archs(self):
+        """@Test: Create an operating system that points at multiple different
+        architectures.
+
+        @Feature: Operating System
+
+        @Assert: The operating system is created and points at the expected
+        architectures.
+        """
+        amount = range(random.randint(3, 5))
```
Comment (abalakh) [inline tests/foreman/api/test_operatingsystem.py]: I'm curios why are we using both `random.randint` and `fauxfactory.gen_integer` in the same module. Mind dropping random dependency and update calls with `gen_integer()`? Not a blocker

File: tests/foreman/api/test_operatingsystem.py [hunk 4]
```diff
@@ -81,17 +188,395 @@ def test_point_to_ptable(self):
...
+    def test_positive_update_description(self):
+        """@Test: Create operating entity providing the initial description,
+        then update that description to another valid one.
+
+        @Feature: Operating System
+
+        @Assert: Operating system entity is created and updated properly
+        """
+        os = entities.OperatingSystem(description=gen_string('utf8')).create()
+        new_desc = gen_string('utf8')
```
Comment (abalakh) [inline tests/foreman/api/test_operatingsystem.py]: > Could you please check different values with subTest here?

**Comment by abalakh** [comment]
Pending comments

File: tests/foreman/api/test_operatingsystem.py (hunk 1 above)
Comment (oshtaier) [inline tests/foreman/api/test_operatingsystem.py]: 1) It is not mine code, so I don'y know what was initial intention
2) I don't like that gen_string('numeric') can generate values like '00001' which will be cut out by further transformation.
3) '999999...' is not hardcoded value, but just max value that can be used, so no problem here. Maybe it doesn't look so nice from aesthetic point of view, but anyway.

However, no serious reasons to not make proposed change :)
Fixed.

File: tests/foreman/api/test_operatingsystem.py (hunk 2 above)
Comment (oshtaier) [inline tests/foreman/api/test_operatingsystem.py]: Don't know why, but added

File: tests/foreman/api/test_operatingsystem.py (hunk 3 above)
Comment (oshtaier) [inline tests/foreman/api/test_operatingsystem.py]: removed `gen_integer` from the code

File: tests/foreman/api/test_operatingsystem.py (hunk 4 above)
Comment (oshtaier) [inline tests/foreman/api/test_operatingsystem.py]: Don't know why, but added

**Comment by oshtaier** [comment]
Addressed all comments. Ready for next round

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/api/test_operatingsystem.py -m test_create_with_minor_1230902
S
----------------------------------------------------------------------
Ran 1 test in 7.916s

OK (SKIP=1)
```
```
nosetests tests/foreman/api/test_operatingsystem.py -m test_positive_create_with_description
.
----------------------------------------------------------------------
Ran 1 test in 15.742s

OK
```
```
nosetests tests/foreman/api/test_operatingsystem.py -m test_positive_update_description
.
----------------------------------------------------------------------
Ran 1 test in 15.632s

OK
```

**Comment by abalakh** [comment]
ACK

**Comment by elyezer** [comment]
NACK, don't remove `from six.moves.http_client import NOT_FOUND` otherwise you will break the compatibillity with python 2 and 3.

File: tests/foreman/api/test_operatingsystem.py [hunk 5]
```diff
@@ -6,11 +6,15 @@
...
+import random
+from fauxfactory import gen_string
+from httplib import NOT_FOUND
 from nailgun import entities
+from requests.exceptions import HTTPError
+from robottelo.constants import OPERATING_SYSTEMS
+from robottelo.datafactory import invalid_values_list, valid_data_list
 from robottelo.decorators import run_only_on, skip_if_bug_open
 from robottelo.test import APITestCase
-from six.moves.http_client import NOT_FOUND
```
Comment (elyezer) [inline tests/foreman/api/test_operatingsystem.py]: This will break python 2 and 3 compatibillity.

Comment (elyezer) [inline tests/foreman/api/test_operatingsystem.py]: Six is now a dependency https://github.com/SatelliteQE/robottelo/blob/master/setup.py#L28

**Comment by oshtaier** [comment]
@elyezer It was my mistake with rebase procedure. Sorry
Fixed.

**Comment by elyezer** [comment]
ACK

**Pull request merged**